api/src/apps/apps.rs

```diff
@@ -894,4 +992,32 @@ Log msg 3 of service-a of app master
 
         Ok(())
     }
+
+    #[test]
+    fn should_delete_apps() -> Result<(), AppsServiceError> {
+        let config = Config::default();
+        let infrastructure = Box::new(Dummy::new());
+        let apps = AppsService::new(config, infrastructure)?;
+
+        let app_name = AppName::from_str("master").unwrap();
+        apps.create_or_update(&app_name, None, &service_configs!("service-a"))?;
+        let deleted_services = apps.delete_app(&app_name)?;
```

Comment:
If I understand correctly, this test does not really cover the complex (racing) case of concurrent requests.
Not sure how difficult it would be to implement that case, as it seems to require multi-threading within the test, so I understand the omission.

Reply:
There is a second test covering this.